### PR TITLE
Fix start of osrm-contract

### PR DESCRIPTION
**Pull request by greenscientist**
osrm-contract expect the file prefix when starting. I don't know how the previous command worked

### CODE REVIEW

**Comment by tahini** [comment]
Really? Here's the output of `osrm-contract --help`:

```
$ osrm-contract --help
Usage: osrm-contract <input.osrm> [options]:
[...]
$ osrm-contract --version
v5.27.1
```

**Comment by greenscientist** [comment]
Yep, that's wrong

**Comment by tahini** [comment]
What's wrong? the osrm-contract help? If so, since when? Was there a recent change? Would this fix fail on some versions?

It works here because the file is called `${mode}.osrm` after extract, so by setting `[mode]` for `<input.osrm>` it looks for `<mode>.osrm`?

**Comment by greenscientist** [comment]
So I don't have a file $mode.osrm anymore here. Might be a new version thing. 
I guess we should test with older version of osrm. 
I based this on the output of osrm-extract which said: 
"To prepare the data for routing, run: ./osrm-contract "/app/examples/runtime/osrm/walking/walking""
Need to go check the code histoire in osrm

**Comment by greenscientist** [comment]
So checking the behavior of the osrm code: 
It takes the parameter and first strip out any known extention before using it as a base path

So basically,   mode.osrm is the same as mode or mode.osm
It does not check that the specific file actually exist, but the rebuild the path to require file base in this base path. 

We can still investigate why we do not have an "mode".osrm file generated anymore. 

Otherwise, we can keep the diff as is, or we can make it so the cmd is with "$mode.osrm" instead of just "$mode".

**Comment by greenscientist** [comment]
commit 21888334dd1e0fbbd25cc62c725eb33bb7968147
Author: Siarhei Fedartsou <siarhei.fedartsou@gmail.com>
Date:   Fri Sep 30 14:29:10 2022 +0200

    Do not generate intermediate .osrm file in osrm-extract. (#6354)

**Comment by greenscientist** [comment]
https://github.com/Project-OSRM/osrm-backend/pull/6354

**Comment by greenscientist** [comment]
it's super confusing and documentation is not well updated yet. 

So I'll change this PR to be mode.osrm

**Comment by tahini** [comment]
With version 5.25, both the current and preceding version work. With 5.27, the contraction works, but the runtime doesn't. Don't you need to update also the osrm-routed path in the OSRMProcessManager.ts file?

**Comment by greenscientist** [comment]
hum, probably yes

**Comment by greenscientist** [comment]
updated, @tahini if you have the chance to test on your side, let me know

**Review by tahini — CHANGES_REQUESTED** [review]

File: packages/chaire-lib-backend/src/utils/processManagers/OSRMProcessManager.ts [hunk 1]
```diff
@@ -130,7 +130,7 @@ const start = function(parameters = {} as any): Promise<any> {
     return new Promise((resolve, reject) => {
         const command = 'osrm-routed';
         const commandArgs = [
-            osrmDirectoryPath + '/*.osrm',
+            osrmDirectoryPath + `/${mode}.osrm`,
```
Comment (tahini) [inline packages/chaire-lib-backend/src/utils/processManagers/OSRMProcessManager.ts:133]: The other comment applies to this line, not the one from OSRMServicePreparation, sorry.

File: packages/chaire-lib-backend/src/utils/processManagers/OSRMServicePreparation.ts [hunk 2]
```diff
@@ -119,7 +119,10 @@ const contract = function(parameters = {} as any): Promise<any> {
     return new Promise((resolve, reject) => {
         console.log(`osrm: contracting osrm data for mode ${mode} from directory ${osrmDirectoryPath}`);
 
-        const osrmProcess = spawn('osrm-contract', ['*.osrm'], {
+        // contract parameter need to match the prefix of the osm file passed to extract
+        // in our case it's the mode name. The .osrm is not generated in the latest osrm-extract
+        // but the name is still accepted as a base path
+        const osrmProcess = spawn('osrm-contract', [`${mode}.osrm`], {
```
[the remark this answers is not in the document]
Comment (greenscientist) [reply packages/chaire-lib-backend/src/utils/processManagers/OSRMServicePreparation.ts:125]: I guess might have a leftover .osrm file locally that made my test pass.

[the remark this answers is not in the document]
Comment (greenscientist) [reply packages/chaire-lib-backend/src/utils/processManagers/OSRMServicePreparation.ts:125]: There's a subtle difference between them. But I've now put the args in a common function.

**Comment by greenscientist** [comment]
I've made sure to erase the old .osrm in my latest test round.

File: packages/chaire-lib-backend/src/utils/processManagers/OSRMProcessManager.ts [hunk 3]
```diff
@@ -118,6 +118,14 @@ const configureAllOsrmServers = async function(startServers = true): Promise<voi
     }
 };
 
+function getOsrmRoutedStartArgs(osrmDirectory: string, mode: string, port: string) : string[] {
```
Comment (tahini) [inline packages/chaire-lib-backend/src/utils/processManagers/OSRMProcessManager.ts:121]: `yarn format` would remove a whitespace before the ':' on this line

**Review by tahini — CHANGES_REQUESTED** [review]
It corresponds to what was working here when tested with both 5.25 and 5.27.1.

Just remove the whitespace so we don't need to add a commit to run `yarn format` later and it's good

**Comment by greenscientist** [comment]
Fixed my yarn format command, so should be good

**Review by tahini — APPROVED** [review]

**Pull request merged**